gap/precompiled_categories/CategoryOfSkeletalFinSetsPrecompiled.gi

```diff
@@ -113,33 +113,31 @@ end
         
 ########
 function ( cat_1, alpha_1 )
-    local hoisted_1_1, hoisted_2_1, hoisted_3_1, hoisted_4_1, hoisted_5_1, hoisted_6_1, deduped_7_1, deduped_8_1, deduped_9_1;
-    deduped_9_1 := Length( Source( alpha_1 ) );
-    deduped_8_1 := Length( Range( alpha_1 ) );
-    deduped_7_1 := [ 1 .. deduped_9_1 ];
-    hoisted_6_1 := deduped_8_1;
+    local hoisted_1_1, hoisted_2_1, hoisted_3_1, hoisted_4_1, hoisted_5_1, hoisted_6_1, hoisted_7_1, hoisted_8_1, hoisted_9_1, deduped_10_1, deduped_11_1, deduped_12_1, deduped_13_1;
+    deduped_13_1 := Length( Source( alpha_1 ) );
+    deduped_12_1 := Length( Range( alpha_1 ) );
+    deduped_11_1 := [ 1 .. deduped_13_1 ];
+    deduped_10_1 := [ 0 .. deduped_13_1 - 1 ];
+    hoisted_9_1 := deduped_12_1;
     hoisted_2_1 := AsList( alpha_1 );
-    hoisted_1_1 := [ 0 .. deduped_9_1 - 1 ];
-    hoisted_5_1 := List( deduped_7_1, function ( i_2 )
+    hoisted_1_1 := deduped_10_1;
+    hoisted_7_1 := List( deduped_11_1, function ( i_2 )
             return hoisted_2_1[1 + hoisted_1_1[i_2]];
         end );
-    hoisted_4_1 := deduped_7_1;
-    hoisted_3_1 := deduped_9_1;
-    return CreateCapCategoryMorphismWithAttributes( cat_1, CreateCapCategoryObjectWithAttributes( cat_1, Length, 1 ), CreateCapCategoryObjectWithAttributes( cat_1, Length, deduped_8_1 ^ deduped_9_1 ), AsList, [ List( [ 0 .. deduped_9_1 ^ deduped_9_1 - 1 ], function ( i_2 )
-                    local hoisted_1_2, hoisted_2_2, hoisted_3_2;
-                    hoisted_1_2 := List( hoisted_1_1, function ( j_3 )
-                            return REM_INT( QUO_INT( i_2, hoisted_3_1 ^ j_3 ), hoisted_3_1 );
-                        end );
-                    hoisted_2_2 := List( hoisted_4_1, function ( i_3 )
-                            return hoisted_1_2[1 + hoisted_1_1[i_3]];
-                        end );
-                    hoisted_3_2 := List( hoisted_4_1, function ( i_3 )
-                            return hoisted_5_1[1 + hoisted_2_2[i_3]];
-                        end );
-                    return Sum( hoisted_1_1, function ( k_3 )
-                            return hoisted_3_2[(1 + k_3)] * hoisted_6_1 ^ k_3;
-                        end );
-                end )[1 + deduped_9_1 * GeometricSumDiff1( deduped_9_1, deduped_9_1 )] ] );
+    hoisted_4_1 := [ 0 .. deduped_13_1 ^ deduped_13_1 - 1 ][1 + deduped_13_1 * GeometricSumDiff1( deduped_13_1, deduped_13_1 )];
```

Comment:
An obvious logic template can simplify this line :)

Reply:
Or maybe it is too late at this point to apply logic templates?

Reply:
Probably :/